app/controllers/course/achievement/condition/achievements_controller.rb

```diff
@@ -0,0 +1,5 @@
+class Course::Achievement::Condition::AchievementsController <
+  Course::Condition::AchievementsController
+
+  include Course::AchievementConditionalConcern
```

Comment:
You mean why there are both inheritance and mixin? The inheritance is for functionalities of the particular condition. The mixin is for routing for a particular conditional.